Makefile

```diff
@@ -24,8 +24,13 @@ export PATH := ./bin:$(PATH)
 export GO111MODULE := on
 export E2E_BINARY
 
+.PHONY: deps
+deps:  ## Download go module dependencies
+	@echo "==> Installing go.mod dependencies..."
+	go mod download
+
 .PHONY: setup
-setup:  ## Install dev tools
+setup: deps ## Install dev tools
```

Comment:
This change right now is breaking our linting step in evergreen. I have not looked at the [evergreen file](https://github.com/mongodb/mongocli/blob/master/build/ci/evergreen.yml) but I guess we are not setting up the go tool chain for that step given we just needed to download a binary and run it and now we would need a go environment 

With that said do we really need this change? We've been relaying on `make build`, `make install` or `make test` to also download dependencies, `golangci-lint` is just a bit special as they don't recommend to install it via deps

If you want to keep this change let's then try to fix evergreen before we merge this